_inc/lib/upgrade-nudge.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Upgrade Nudge Library
+ *
+ * Display a plan upgrade nudge on the frontend
+ */
+
+class Jetpack_Upgrade_Nudge {
+
+	/**
+	 * Return a message telling the user to upgrade to enable the block.
+	 *
+	 * @since 7.6.0
+	 *
+	 * @return string The message telling the user to upgrade
+	 */
+	public static function get_upgrade_message() {
+		$support_url = ( defined( 'IS_WPCOM' ) && IS_WPCOM )
+		? 'https://support.wordpress.com/simple-payments/'
+		: 'https://jetpack.com/support/simple-payment-button/';
+
+		return sprintf(
+			wp_kses(
+				__( 'Your plan doesn\'t include Simple Payments. <a href="%s" rel="noopener noreferrer" target="_blank">Learn more and upgrade</a>.', 'jetpack' ),
+				array( 'a' => array( 'href' => array(), 'rel' => array(), 'target' => array() ) )
+			),
+			esc_url( $support_url )
+		);
+	}
+
+	/**
+	 * Return a message telling the user to upgrade to enable the block.
+	 *
+	 * @since 7.6.0
+	 *
+	 * @return string The message telling the user to upgrade
+	 */
+	public static function get_upgrade_nudge( $plan_name = 'Premium' ) {
+		$title = sprintf( __( 'This block is available under the %1$s Plan.', 'jetpack' ),
+			$plan_name
+		);
+		$message = __( 'It will be hidden from site visitors until you upgrade.', 'jetpack' );
+		$button_label = __( 'Upgrade', 'jetpack' );
+
+		// TODO: Make button work
+
+		return <<<EOF
```

Comment:
You might be saved from the rabbit hole by abandoning the goal of getting the same markup and shared CSS from the editor to work in the frontend, reasons I'm suggesting that are:

- it's a complex process that isn't likely utilized elsewhere than these nudges
- CSS and `editor-warning` is heavily geared towards working in a block-editor context; it would be very easy to miss this frontend nudge breaking because of changes in upgrade nudge made for the editor or worse, changes in Gutenberg itself.
- You'd have to enqueue all `wp-block-editor`/`wp-editor`/`wp-components`/`wp-editor-font` ([ref](https://github.com/WordPress/gutenberg/blob/cf1da64370c209b25bb005c44083097a6137a119/lib/client-assets.php#L311-L325)) styles which is a lot of CSS and none of it is scoped (things like `.components-button` or `.DayPicker`)

Since this upgrade nudge isn't probably seen very often (primary way of seeing the nudge would be in the editor), I'd suggest sticking to something simple and straightforward (HTML markup wise), similar to what current simple payments already has.

There are tons of good reasons to stick to one source of truth tho and those might very well outweigh risks. What do you think?

This same convo might've already happened somewhere else, I'm only catching up. ;-)